Remove NDCube.world_axis_physical_types.

DanRyanIrish opened the pull request:
Use version attached to wcs instead.  Addresses #275

pep8speaks commented:
Hello @DanRyanIrish! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:









* In the file [`ndcube/tests/test_ndcollection.py`](https://github.com/sunpy/ndcube/blob/8e2a0a3e4c2e031516a62b25a69f1205f1431320/ndcube/tests/test_ndcollection.py):

> [Line 142:1](https://github.com/sunpy/ndcube/blob/8e2a0a3e4c2e031516a62b25a69f1205f1431320/ndcube/tests/test_ndcollection.py#L142): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

DanRyanIrish commented:
@Cadair, as well as removing ```world_axis_physical_types```, I've changed the output type of ```NDCollection.aligned_world_axis_physical_types``` to a ```list``` to be consistent with ```WCS.world_axis_physical types.  This came about because I had to refactor that method as it relied on ```NDCube.world_axis_physical_types.  Anyway, I've added to changelogs, one named ```302.breaking.rst``` due to the output type change.  Is this how it should be done?

Cadair commented on changelog/302.breaking.rst:
```diff
@@ -0,0 +1 @@
+Change type of output of NDCollection.aligned_world_axis_physical_types from tuple to list. This is to be consistent with output of astropy.wcs.WCS.world_axis_physical_types.
```
```suggestion
Change type of output of `.NDCollection.aligned_world_axis_physical_types` from tuple to list. This is to be consistent with output of `astropy.wcs.WCS.world_axis_physical_types`.
```

Cadair commented on changelog/302.removal.rst:
```diff
@@ -0,0 +1 @@
+Remove world_axis_physical_types methods from NDCube and NDCubeSequence.
```
```suggestion
Remove ``world_axis_physical_types`` methods from `.NDCube` and  `.NDCubeSequence`.
```
